platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transients/field/WorkflowUserField.java

```diff
@@ -61,22 +63,65 @@ private List<String> getUserTaskAgeStrings(R row) {
 			.collect(Collectors.toList());
 	}
 
-	private Map<AGUser, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
+	private Map<UserAndDelegationUser, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
 
 		AGWorkflowTask.CREATION_TRANSACTION.prefetchAll(tasks);
 
-		TreeMap<AGUser, DayTime> userTaskCreationMap = new TreeMap<>();
+		Map<UserAndDelegationUser, DayTime> userTaskCreationMap = new HashMap<>();
 		for (AGWorkflowTask task: tasks) {
 			Optional<AGTransaction> creationTransaction = Optional.ofNullable(AGWorkflowTask.CREATION_TRANSACTION.getValue(task));
-			if (creationTransaction.isPresent()) {
-				userTaskCreationMap.merge(task.getUser(), creationTransaction.get().getAt(), DayTime::min);
+
+			if (creationTransaction.isPresent() && !task.hasActiveDelegation()) {
```

Comment:
Checking `hasActiveDelegation` twice in this if-else construct seems odd to me. How about this structure?

```
if (task.hasActiveDelegation()) {
	// ...big code block...
} else {
	if(creationTransaction.isPresent()) {
		// ...small code block...
	}
}

Reply:
jeb, i switched to this way